boot/freeldr/freeldr/lib/fs/fat.c

```diff
@@ -1590,10 +1601,12 @@ const DEVVTBL* FatMount(ULONG DeviceId)
     //
     // Check if BootSector is valid. If no, return early
     //
+    IsFatX = RtlEqualMemory(BootSectorX->FileSystemType, "FATX", 4);
+
     if (!RtlEqualMemory(BootSector->FileSystemType, "FAT12   ", 8) &&
         !RtlEqualMemory(BootSector->FileSystemType, "FAT16   ", 8) &&
         !RtlEqualMemory(BootSector32->FileSystemType, "FAT32   ", 8) &&
-        !RtlEqualMemory(BootSectorX->FileSystemType, "FATX", 4))
+        !IsFatX)
```

Comment:
Better fix is to keep the original code here, and in the last return statement for the function table, you do instead:
`return (ISFATX(Volume->FatType) ? &FatXFuncTable : &FatFuncTable);`